### PR TITLE
Purge verified test requests with wiremock

### PR DESCRIPTION
**Pull request by SlimaneAmar**
Signed-off-by: Slimane AMAR <slimane.amar@rte-france.com>

### CODE REVIEW

File: src/test/java/org/gridsuite/study/server/NetworkModificationTest.java [hunk 1]
```diff
@@ -2429,8 +2418,8 @@ public void tearDown() throws IOException {
         TestUtils.assertQueuesEmptyThenClear(List.of(studyUpdateDestination), output);
 
         try {
-            // it returns an exception if a request was not matched by wireMock, but does not complain if it was not verified by 'verify'
```
Comment (flomillot) [inline src/test/java/org/gridsuite/study/server/NetworkModificationTest.java:2432]: Why removing the comment ? If so, I think you should explain a bit more below

File: src/test/java/org/gridsuite/study/server/NetworkModificationTest.java [hunk 2]
```diff
@@ -2444,79 +2433,108 @@ public void tearDown() throws IOException {
...
+    private void verifyPutRequestWithUrlMatching(UUID stubId, String url, Map<String, StringValuePattern> queryParams, String body) {
+        verifyPutRequest(stubId, url, true, queryParams, body);
     }
 
-    private void verifyPutWithBody(String requestBody) {
-        wireMock.verify(1, WireMock.putRequestedFor(WireMock.urlPathEqualTo(
-                        "/v1/network-modifications/" + MODIFICATION_UUID))
-                .withRequestBody(WireMock.equalToJson(requestBody)));
+    private void verifyPutRequest(UUID stubId, String url, boolean urlMatching, Map<String, StringValuePattern> queryParams, String body) {
+        RequestPatternBuilder requestBuilder = urlMatching ? WireMock.putRequestedFor(WireMock.urlMatching(url)) : WireMock.putRequestedFor(WireMock.urlPathEqualTo(url));
```
Comment (flomillot) [inline src/test/java/org/gridsuite/study/server/NetworkModificationTest.java:2517]:
```suggestion
        RequestPatternBuilder requestBuilder = urlMatching ? WireMock.putRequestedFor(WireMock.urlPathMatching(url)) : WireMock.putRequestedFor(WireMock.urlPathEqualTo(url));
```

Comment (flomillot) [reply src/test/java/org/gridsuite/study/server/NetworkModificationTest.java:2517]: For the 2 others too

File: src/test/java/org/gridsuite/study/server/NetworkModificationTest.java [hunk 3]
```diff
@@ -2444,79 +2433,108 @@ public void tearDown() throws IOException {
...
-                .withQueryParam("networkUuid", WireMock.equalTo(NETWORK_UUID_STRING))
-                .withQueryParam("groupUuid", WireMock.matching(".*"))
-                .withRequestBody(WireMock.equalToJson(requestBody)));
+    private void verifyPostRequest(UUID stubId, String url, Map<String, StringValuePattern> queryParams) {
+        verifyPostRequest(stubId, url, false, queryParams, null);
     }
 
-    private void verifyPostWithBodyAndVariant(String requestBody, String variantId) {
-        verifyPostWithBodyAndVariant(1, requestBody, variantId);
+    private void verifyPostRequest(UUID stubId, String url, boolean urlMatching, Map<String, StringValuePattern> queryParams, String body) {
```
Comment (flomillot) [inline src/test/java/org/gridsuite/study/server/NetworkModificationTest.java:2507]: urlMatching -> regex or regexMatching ?

Comment (flomillot) [reply src/test/java/org/gridsuite/study/server/NetworkModificationTest.java:2507]: url -> urlPath ?

File: src/test/java/org/gridsuite/study/server/NetworkModificationTest.java [hunk 4]
```diff
@@ -2444,79 +2433,108 @@ public void tearDown() throws IOException {
...
+        queryParams.forEach((k, v) -> requestBuilder.withQueryParam(k, v));
+        if (body != null) {
+            requestBuilder.withRequestBody(WireMock.equalToJson(body));
+        }
+        wireMock.verify(1, requestBuilder);
+        removeRequestForStub(stubId);
+    }
+
+    private void removeRequestForStub(UUID stubId) {
+        List<ServeEvent> serveEvents = WireMock.getAllServeEvents(ServeEventQuery.forStubMapping(stubId));
```
Comment (flomillot) [inline src/test/java/org/gridsuite/study/server/NetworkModificationTest.java:2536]:
```suggestion
        List<ServeEvent> serveEvents = wireMock.getServeEvents(ServeEventQuery.forStubMapping(stubId)).getServeEvents();
```

File: src/test/java/org/gridsuite/study/server/NetworkModificationTest.java [hunk 5]
```diff
@@ -2444,79 +2433,108 @@ public void tearDown() throws IOException {
...
+        assertEquals(1, serveEvents.size());
+        WireMock.removeServeEvent(serveEvents.get(0).getId());
```
Comment (flomillot) [inline src/test/java/org/gridsuite/study/server/NetworkModificationTest.java:2538]:
```suggestion
        wireMock.removeServeEvent(serveEvents.get(0).getId());
```

File: src/test/java/org/gridsuite/study/server/NetworkModificationTest.java (hunk 2 above)
Comment (SlimaneAmar) [reply src/test/java/org/gridsuite/study/server/NetworkModificationTest.java:2517]: ?

File: src/test/java/org/gridsuite/study/server/NetworkModificationTest.java (hunk 5 above)
Comment (SlimaneAmar) [reply src/test/java/org/gridsuite/study/server/NetworkModificationTest.java:2538]: Done

File: src/test/java/org/gridsuite/study/server/NetworkModificationTest.java (hunk 4 above)
Comment (SlimaneAmar) [reply src/test/java/org/gridsuite/study/server/NetworkModificationTest.java:2536]: Done

File: src/test/java/org/gridsuite/study/server/NetworkModificationTest.java (hunk 3 above)
Comment (SlimaneAmar) [reply src/test/java/org/gridsuite/study/server/NetworkModificationTest.java:2507]: Done

File: src/test/java/org/gridsuite/study/server/NetworkModificationTest.java (hunk 2 above)
Comment (SlimaneAmar) [reply src/test/java/org/gridsuite/study/server/NetworkModificationTest.java:2517]: Done

File: src/test/java/org/gridsuite/study/server/NetworkModificationTest.java (hunk 1 above)
Comment (SlimaneAmar) [reply src/test/java/org/gridsuite/study/server/NetworkModificationTest.java:2432]: Done

File: src/test/java/org/gridsuite/study/server/NetworkModificationTest.java [hunk 6]
```diff
@@ -199,6 +208,7 @@ public void setup() throws IOException {
         // Start the mock servers
         server.start();
         wireMock.start();
+        configureFor("localhost", wireMock.port());
```
[the remark this answers is not in the document]
Comment (SlimaneAmar) [reply src/test/java/org/gridsuite/study/server/NetworkModificationTest.java:211]: Remove it

[the remark this answers is not in the document]
Comment (flomillot) [reply src/test/java/org/gridsuite/study/server/NetworkModificationTest.java:211]: I don't see the commit, did you push it ?

File: src/test/java/org/gridsuite/study/server/NetworkModificationTest.java (hunk 2 above)
Comment (SlimaneAmar) [reply src/test/java/org/gridsuite/study/server/NetworkModificationTest.java:2517]: it was already like that !

Comment (flomillot) [reply src/test/java/org/gridsuite/study/server/NetworkModificationTest.java]: I guess some params are missing, we can't fix it ?

Comment (flomillot) [reply src/test/java/org/gridsuite/study/server/NetworkModificationTest.java]: I'm checking

Comment (flomillot) [reply src/test/java/org/gridsuite/study/server/NetworkModificationTest.java]: Fixed

**Review by flomillot — APPROVED** [review]

**Pull request merged**